course/include/isometry.h

```diff
@@ -0,0 +1,84 @@
+#ifndef ISOMETRY_H
+#define ISOMETRY_H
+
+#include <cstdint>
+#include <iostream>
+#include <ostream>
+
+namespace ekumen {
+namespace math {
+
+class Vector3{
+    public:
+        // Constructors
+        Vector3(double x = 0, double y = 0, double z = 0) : x_{x}, y_{y}, z_{z} {};
+        Vector3(const Vector3& v) : Vector3(v.x_, v.y_, v.z_) {};
+        Vector3(const std::initializer_list<double>& l) {
+            if(l.size() != 3) {
+                throw "Invalid initializer list size";
+            }
+            std::initializer_list<double>::iterator it = l.begin();
+            x_ = *it++;
+            y_ = *it++;
+            z_ = *it;
+        }
+
+        // Operators
+        double& operator [] (const int);
+        const double& operator [] (const int) const;
+        Vector3 operator + (const Vector3&) const;
+        Vector3 operator - (const Vector3&) const;
+        Vector3 operator * (const Vector3&) const;
+        Vector3 operator * (const double&) const;
+        Vector3 operator / (const Vector3&) const;
+        Vector3& operator = (const Vector3&);
+        Vector3& operator += (const Vector3&);
+        Vector3& operator -= (const Vector3&);
+        Vector3& operator *= (const Vector3&);
+        Vector3& operator /= (const Vector3&);
+        bool operator == (const Vector3&) const;
+        bool operator != (const Vector3&) const;
+        bool operator == (const std::initializer_list<double>& list) const;
+        bool operator != (const std::initializer_list<double>& list) const;
+
+        friend Vector3 operator * (const double d, const Vector3& v) {
+            return Vector3(v.x_ * d, v.y_ * d, v.z_ * d);
+        }
+
+        friend std::ostream& operator << (std::ostream& os, const Vector3& v) {
+            os << std::string("(x: ") << v.x() << ", y: " << v.y() << ", z: " << v.z() << ")";
+            return os;
+        }
+
+        // Getters
+        double x() const {return x_;};
+        double y() const {return y_;};
+        double z() const {return z_;};
```

Comment:
Fine. Could also have returned `const double &`

Reply:
But given the following code:

`Vector3 v;
double a = v.x() +3 ;`

Wouldn't it create an error from the compiler if it is returning a const?

Reply:
I assume you meant the vector to be const:
```
const Vector3 v;
double a = v.x() +3 ;
```
In that case, no, because you are not changing the value of x(), you're using it in an addition that creates a new object as a result.

Reply:
You are right. Thanks!